packages/realm-react/src/AppProvider.tsx

```diff
@@ -41,32 +41,24 @@ type AppProviderProps = Realm.AppConfiguration & {
 export const AppProvider: React.FC<AppProviderProps> = ({ children, appRef, ...appProps }) => {
   const configuration = useRef<Realm.AppConfiguration>(appProps);
 
-  const [app, setApp] = useState<Realm.App>(new Realm.App(configuration.current));
-
-  // We increment `configVersion` when a config override passed as a prop
-  // changes, which triggers a `useEffect` to overwrite the current App with the
-  // new config
-  const [configVersion, setConfigVersion] = useState(0);
-
-  useEffect(() => {
-    if (!isEqual(appProps, configuration.current)) {
-      configuration.current = appProps;
-      setConfigVersion((x) => x + 1);
-    }
-  }, [appProps]);
+  const [app, setApp] = useState<Realm.App>(() => new Realm.App(configuration.current));
 
   // Support for a possible change in configuration
-  useEffect(() => {
+  if (!isEqual(appProps, configuration.current)) {
+    configuration.current = appProps;
+
     try {
-      const app = new Realm.App(configuration.current);
-      setApp(app);
-      if (appRef) {
-        appRef.current = app;
-      }
+      setApp(new Realm.App(configuration.current));
     } catch (err) {
       console.error(err);
     }
-  }, [configVersion, setApp]);
+  }
+
+  useEffect(() => {
```

Comment:
I think this `useEffect` could also go away and the logic moved under line 51.  Thoughts?

Reply:
Thanks for taking a look at this @takameyer! Your asking this made me realize this *technically* should be `useLayoutEffect`, so I'll push that update. This is a nuanced one, especially with React's concurrent mode semantics coming to React Native. Refs are typically updated by React right before running layout effects, not during render since a render pass can be thrown away. While it's *fine* (ish) to update a ref you "own" during render (though that can lead to trouble too), refs you don't own should be updated as side effects because a render can be thrown away with concurrent mode. Also, another benefit here is that if `appRef` itself changes for some reason, then it'll get updated accordingly. 😀

Reply:
Now that you mention that, I will have to give the generated `RealmProvider` a look as well.  This could most likely also benefit from `useLayoutEffect` when binding the `realmRef`.

Reply:
Hi @takameyer, checking to see if this can be merged. I see a bunch of CI jobs are failing in the same way that appears to be unrelated...

```
An unexpected error occurred: Input required and not supplied: projectId
```

Reply:
@appden The fix is now released in [0.4.3](https://github.com/realm/realm-js/releases/tag/realm-react-v0.4.3)